### DISCUSSION TITLE
Rendering on machines without displays (eg. server)

### DISCUSSION THREAD

**Discussion by sibocw** [NeLy-EPFL/flygym#104] [General]
If you are using a machine without a display (e.g. server, cluster, remotely accessed workstation), you will need to change the renderer to EGL. See [this section of the installation instruction](https://neuromechfly.org/installation.html#special-notes-for-rendering-on-machines-without-a-display) for details. If it still doesn't work, check [this link](https://pytorch.org/rl/reference/generated/knowledge_base/MUJOCO_INSTALLATION.html#prerequisite-for-rendering-all-mujoco-versions) for further information on dependencies.

If you encounter `FatalError: gladLoadGL error` upon calling `.render()`, this is likely the reason.

**Comment by Lunamos** [comment]
I encountered a similar problem and found a solution.

If you are running the examples in the notebook, you will still get `FatalError: gladLoadGL` error if you change the renderer to EGL.

This seems to be a problem with jupyter notebook. As long as you run it in a normal `.py` file and set the renderer to EGL correctly, it will render normally.
